syntax = "proto3";

package docssummary;

message DocsSummaryRequest {
  string user_id = 1;
  string chunk = 2;  // 요약할 텍스트 조각(한 문단 등)
}

message DocsSummaryResponse {
  string line = 1;  // 	요약된 문장 또는 결과 문자열
}

service DocsSummaryService {
  rpc SummarizeStream (stream DocsSummaryRequest) returns (stream DocsSummaryResponse);
}

// rpc SummarizeStream : 메서드 이름(클라이언트에서 호출됨)
// stream DocsSummaryRequest : 클라이언트가 여러 요청(chunk)을 스트림으로 보냄
// stream DocsSummaryResponse : 서버도 여러 응답(line)을 스트림으로 전환